pkg/resource/rule_groups_namespace/hooks.go

```diff
@@ -0,0 +1,345 @@
+// of being deleted
+func ruleGroupsNamespaceDeleting(r *resource) bool {
+	if r.ko.Status.Status == nil {
+		return false
+	}
+	ws := *r.ko.Status.Status.StatusCode
+	return ws == string(svcapitypes.RuleGroupsNamespaceStatusCode_DELETING)
+}
+
+// ruleGroupsNamespaceUpdating returns true if the supplied rule groups is in the process
+// of being updated
+func ruleGroupsNamespaceUpdating(r *resource) bool {
+	if r.ko.Status.Status == nil {
+		return false
+	}
+	ws := *r.ko.Status.Status.StatusCode
+	return ws == string(svcapitypes.RuleGroupsNamespaceStatusCode_UPDATING)
+}
+
+// ruleGroupsNamespaceHasTerminalStatus returns whether the supplied rule groups namespace is in a
+// terminal state
+func ruleGroupsNamespaceHasTerminalStatus(r *resource) bool {
+	if r.ko.Status.Status.StatusCode == nil {
+		return false
+	}
+	ts := *r.ko.Status.Status.StatusCode
+	for _, s := range TerminalStatuses {
+		if ts == string(s) {
+			return true
+		}
+	}
+	return false
+}
+
+// customUpdateRuleGroupsNamespace patches each of the resource properties in the backend AWS
+// service API and returns a new resource with updated fields.
+func (rm *resourceManager) customUpdateRuleGroupsNamespace(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdateRuleGroupsNamespace")
+	defer exit(err)
+
+	// Check if the state is being currently created, updated or deleted.
+	// If it is, then requeue because we can't update while it is in those states.
+	var sc string = ""
+	if latest.ko.Status.Status != nil {
+		sc = *latest.ko.Status.Status.StatusCode
+	}
+	switch sc {
+	case string(svcapitypes.RuleGroupsNamespaceStatusCode_DELETING):
+		return desired, requeueWaitWhileDeleting
+	case string(svcapitypes.RuleGroupsNamespaceStatusCode_UPDATING):
+		return desired, requeueWaitWhileUpdating
+	case string(svcapitypes.RuleGroupsNamespaceStatusCode_CREATING):
+		return desired, requeueWaitWhileCreating
+	}
+
+	if ruleGroupsNamespaceHasTerminalStatus(latest) {
+		msg := "Rule Groups Namespace is in '" + *latest.ko.Status.Status.StatusCode + "' status"
+		ackcondition.SetTerminal(desired, corev1.ConditionTrue, &msg, nil)
+		ackcondition.SetSynced(desired, corev1.ConditionTrue, nil, nil)
+		return desired, nil
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	rm.setStatusDefaults(ko)
+	if delta.DifferentAt("Spec.Tags") {
+		err = rm.updateRuleGroupsNamespaceTags(ctx, latest, desired)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if delta.DifferentAt("Spec.Configuration") {
+		updatedResource, err := rm.updateRuleGroupsNamespace(ctx, desired)
+		if err != nil {
+			return nil, err
+		}
+		if updatedResource != nil {
+			return updatedResource, nil
+		}
```

Comment:
I think `updatedResource` will always be non nil if err is nil

Reply:
Good catch